hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/AzureNativeFileSystemStore.java

```diff
@@ -1858,7 +1881,13 @@ public void storeEmptyLinkFile(String key, String tempBlobKey,
       CloudBlobWrapper blob = getBlobReference(key);
       storePermissionStatus(blob, permissionStatus);
       storeLinkAttribute(blob, tempBlobKey);
-      openOutputStream(blob).close();
+      if (eTag != null) {
+        AccessCondition accessCondition = new AccessCondition();
+        accessCondition.setIfMatch(eTag);
+        openOutputStream(blob, accessCondition).close();
```

Comment:
We can get etag here and pass that till the point where we want to rename. No need of metadata call is required.

`blob.getBlob().getProperties().getEtag();` will help. why?
`blob` here is CloudBlockBlobWrapper. `blob.getBlob()` is CloudBlockBlob.
Now, in `CloudBlockBlob.commitBlockListImpl.preProcessResponse`, etag and LMT will be updated in the CloudBlockBlob obj.